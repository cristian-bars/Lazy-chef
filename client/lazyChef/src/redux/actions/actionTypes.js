export default {
  LOAD_RECIPES: 'LOAD_RECIPES',
  DELETE_RECIPE: 'DELETE_RECIPE',
  ADD_RECIPE: 'ADD_RECIPE',
  UPDATE_RECIPE: 'UPDATE_RECIPE',
  LOAD_RECIPE: 'LOAD_RECIPE',

  LOAD_USERS: 'LOAD_USERS',
  DELETE_USER: 'DELETE_USER',
  ADD_USER: 'ADD_USER',
  UPDATE_USER: 'UPDATE_USER',
  LOAD_USER: 'LOAD_USER',
};
