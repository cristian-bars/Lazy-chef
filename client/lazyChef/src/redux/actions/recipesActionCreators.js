import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/recipes';
const url = 'http://192.168.0.29:2022/recipes';
//const url = 'http://192.168.1.130:2022/recipes';

export function loadRecipes() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_RECIPES,
        recipes: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPES_ERROR',
      });
    }
  };
}

export function addRecipe(recipe) {
  return {
    type: actionTypes.ADD_RECIPE,
    recipe,
  };
}

export function deleteRecipe(recipeId) {
  return async dispatch => {
    await axios.delete(`${url}/${recipeId.id}`);
    dispatch({
      type: actionTypes.DELETE_RECIPE,
      recipeId,
    });
  };
}

export function updateRecipe(recipe) {
  return {
    type: actionTypes.UPDATE_RECIPE,
    recipe,
  };
}

export function loadRecipe(recipe) {
  return {
    type: actionTypes.LOAD_RECIPE,
    recipe,
  };
}

export function getRecipeById(recipeId) {
  console.log(`${url}/${recipeId}`);
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/${recipeId}`);
      dispatch({
        type: actionTypes.LOAD_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPE_ERROR',
      });
    }
  };
}
