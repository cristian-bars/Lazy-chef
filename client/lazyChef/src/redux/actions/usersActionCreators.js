import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/users';
const url = 'http://192.168.0.29:2022';
//const url = process.env.REACT_APP_PORT;

export function loadUsers() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_USERS,
        users: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}

export function addUsers(user) {
  return {
    type: actionTypes.ADD_USER,
    user,
  };
}

export function deleteUser(userId) {
  return async dispatch => {
    await axios.delete(`${url}/${userId.id}`);
    dispatch({
      type: actionTypes.DELETE_USER,
      userId,
    });
  };
}

export function updateUser(user) {
  return {
    type: actionTypes.UPDATE_USER,
    user,
  };
}

export function loadUser(user) {
  return {
    type: actionTypes.LOAD_USER,
    user,
  };
}

export function getUserById(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/login`, user);
      dispatch({
        type: actionTypes.LOAD_USER,
        userToken: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}
