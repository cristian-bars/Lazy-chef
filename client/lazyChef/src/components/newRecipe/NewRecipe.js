/* eslint-disable no-shadow */
import React, {useEffect, useState} from 'react';
import {connect} from 'react-redux';
import {View, TouchableOpacity, Text, TextInput, Image} from 'react-native';
import generalStyles from '../../../generalStyles';
import {addRecipe} from '../../redux/actions/recipesActionCreators';
import {updateUser} from '../../redux/actions/usersActionCreators';
import styles from './newRecipeStyles';

const NewRecipe = ({userAcces, dispatch, navigation, recipes}) => {
  useEffect(() => {
    userAcces.token && navigation.navigate('User');
  }, [userAcces, navigation]);

  let [title, showTitle] = useState('');
  let [description, showDescription] = useState('');
  const author = userAcces.user.name;
  const datePublished = new Date();

  const addNewRecipe = () => {
    dispatch(addRecipe({title, description, author, datePublished}));
  };

  const goBack = () => {
    const userRecipesIds = userAcces.user.recipesCreated;
    userRecipesIds.push(recipes[recipes.length - 1]._id);
    console.log(userAcces.user);
    dispatch(
      updateUser({
        id: userAcces.user._id,
        recipesCreated: userAcces.user.recipesCreated,
        bearerToken: userAcces.token,
      }),
    );
    navigation.navigate('User');
  };

  return (
    <View style={styles.newRecipeContainer}>
      <TouchableOpacity
        style={[generalStyles.roundButton, generalStyles.roundBackButton]}
        onPress={() => goBack()}>
        <Image
          style={generalStyles.backImage}
          source={require('../../img/back.png')}
        />
      </TouchableOpacity>
      <TextInput
        style={generalStyles.formLogin}
        placeholder="Title"
        onChangeText={title => showTitle(title)}
      />
      <TextInput
        style={generalStyles.formLogin}
        placeholder="Description"
        onChangeText={description => showDescription(description)}
      />
      <TouchableOpacity style={generalStyles.button}>
        <Text style={generalStyles.baseText} onPress={addNewRecipe}>
          AÃ±adir
        </Text>
      </TouchableOpacity>
    </View>
  );
};

function mapStateToProps({userAcces, recipes}) {
  return {
    userAcces,
    recipes,
  };
}

export default connect(mapStateToProps)(NewRecipe);
