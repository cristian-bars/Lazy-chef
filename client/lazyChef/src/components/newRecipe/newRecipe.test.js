import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Store from 'redux-mock-store';
import * as actions from '../../redux/actions/recipesActionCreators';
import {Provider} from 'react-redux';
import NewRecipe from './index';

describe('Given a newRecipe component', () => {
  const mockStore = Store();
  let myStore;
  let navigation;

  beforeEach(() => {
    jest.spyOn(actions, 'addRecipe').mockReturnValueOnce({type: ''});
    myStore = mockStore({
      userAcces: {user: {name: 'Cristian'}},
      recipe: {title: 'meat'},
    });
    navigation = {
      navigate: jest.fn(),
      goBack: jest.fn(),
    };
  });
  describe('When is render', () => {
    describe('And all is correct', () => {
      test('Then is matched to snapshot', () => {
        const tree = render(
          <Provider store={myStore}>
            <NewRecipe navigation={navigation} />
          </Provider>,
        );

        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('And user have token', () => {
    test('Then navigation is invoked ', () => {
      myStore = mockStore({
        userAcces: {
          user: {name: 'Cristian'},
          token: '1234abc',
        },
      });

      render(
        <Provider store={myStore}>
          <NewRecipe navigation={navigation} />
        </Provider>,
      );

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  describe('When add recipe button is pressed', () => {
    test('Then function addRecipe is called', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <NewRecipe navigation={navigation} />
        </Provider>,
      );
      const addRecipe = getByTestId('addRecipe');
      fireEvent.press(addRecipe);

      expect(actions.addRecipe).toHaveBeenCalled();
    });
  });

  describe('When titleInput is typing with Ensalada', () => {
    test('Then titleInput value is Ensalada', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <NewRecipe navigation={navigation} />
        </Provider>,
      );
      const titleInput = getByTestId('titleInput');
      const title = 'Ensalada';
      fireEvent.changeText(titleInput, title);

      expect(titleInput.props.value).toBe(title);
    });
  });

  describe('When descriptionInput is typing with Fresca', () => {
    test('Then descriptionInput value is Fresca', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <NewRecipe navigation={navigation} />
        </Provider>,
      );
      const descriptionInput = getByTestId('descriptionInput');
      const description = 'Fresca';
      fireEvent.changeText(descriptionInput, description);

      expect(descriptionInput.props.value).toBe(description);
    });
  });
});
