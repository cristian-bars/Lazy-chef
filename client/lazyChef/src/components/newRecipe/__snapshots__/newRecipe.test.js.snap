// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a newRecipe component When is render And all is correct Then is matched to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D3E0FF",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
      "paddingTop": 50,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 10,
        "left": 20,
        "opacity": 1,
        "padding": 20,
        "position": "absolute",
        "top": 20,
        "width": 10,
        "zIndex": 1,
      }
    }
    testID="goBack"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/back.png",
        }
      }
      style={
        Object {
          "height": 20,
          "left": -12,
          "position": "relative",
          "top": -10,
          "width": 20,
        }
      }
    />
  </View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Title"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    testID="titleInput"
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Description"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    testID="descriptionInput"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 30,
        "opacity": 1,
        "width": 300,
      }
    }
    testID="addRecipe"
  >
    <Text
      style={
        Object {
          "fontFamily": "Cochin",
          "fontSize": 25,
        }
      }
    >
      AÃ±adir
    </Text>
  </View>
</View>
`;
